### PR TITLE
Fix 3.10 schema for MS SQL

### LINKED ISSUE #2884: MS SQL foreign key issue
Checked recent changes on MS SQL and got an error (not surprised):
````
SEVERE 2/8/17 1:01 PM: liquibase: schema/changelog-master.xml: changelog-3.10::changelog-3.10::author: Change Set changelog-3.10::changelog-3.10::author failed.  Error: Введение ограничения внешнего ключа (FOREIGN KEY) "fk_user_user_manageduserid" для таблицы "user_user" может привести к появлению циклов или множественных каскадных путей. Укажите ON DELETE NO ACTION или ON UPDATE NO ACTION либо измените другие ограничения внешнего ключа (FOREIGN KEY). [Failed SQL: ALTER TABLE [dbo].[user_user] ADD CONSTRAINT [fk_user_user_manageduserid] FOREIGN KEY ([manageduserid]) REFERENCES [dbo].[users] ([id]) ON DELETE CASCADE]
````

I'm going to add "-notmssql" changeset like in 3.7.
Also I'm thinking of improving  `database.deleteUser` request to cleanup `users_users` table after it.
Is that appropriate way?

### PR DESCRIPTION
**Pull request by Abyss777**
Replaced `fk_user_user_manageduserid` in MS SQL with trigger (not sure in trigger name)

Also, you were right, we need drop `defaultValue` before add, because MS SQL do not have `SET DEFAULT` only `ADD DEFAULT`

fix #2884

### CODE REVIEW

File: schema/changelog-3.10.xml [hunk 1]
```diff
@@ -62,17 +62,43 @@
...
     </update>
+    <dropDefaultValue tableName="users" columnName="devicelimit" />
     <addDefaultValue tableName="users" columnName="devicelimit" defaultValueNumeric="-1" />
 
     <addColumn tableName="users">
       <column name="devicereadonly" type="BOOLEAN" defaultValueBoolean="false" />
     </addColumn>
 
   </changeSet>
+  <changeSet author="author" id="changelog-3.10-notmssql">
```
Comment (tananaev) [inline schema/changelog-3.10.xml]: Please add space here.

File: schema/changelog-3.10.xml [hunk 2]
```diff
@@ -62,17 +62,43 @@
...
+    <preConditions onFail="MARK_RAN">
+      <not>
+        <dbms type="mssql" />
+      </not>
+    </preConditions>
+    <addForeignKeyConstraint baseTableName="user_user" baseColumnNames="manageduserid" constraintName="fk_user_user_manageduserid" referencedTableName="users" referencedColumnNames="id" onDelete="CASCADE" />
+
+  </changeSet>
+
+  <changeSet author="author" id="changelog-3.10-mssql">
```
Comment (tananaev) [inline schema/changelog-3.10.xml]: Should we do same thing for groups?

File: schema/changelog-3.10.xml [hunk 3]
```diff
@@ -62,17 +62,43 @@
     </createTable>
 
     <addForeignKeyConstraint baseTableName="user_user" baseColumnNames="userid" constraintName="fk_user_user_userid" referencedTableName="users" referencedColumnNames="id" onDelete="CASCADE" />
-    <addForeignKeyConstraint baseTableName="user_user" baseColumnNames="manageduserid" constraintName="fk_user_user_manageduserid" referencedTableName="users" referencedColumnNames="id" onDelete="CASCADE" />
 
     <update tableName="users">
         <column name="devicelimit" valueNumeric="-1" />
         <where>devicelimit = 0</where>
     </update>
+    <dropDefaultValue tableName="users" columnName="devicelimit" />
```
Comment (tananaev) [inline schema/changelog-3.10.xml]: So, it is required? I remember I asked you about it.

Comment (Abyss777) [inline schema/changelog-3.10.xml]: Unfortunately required. For MS SQL.
My bad, liquibase documentation confused me.

File: schema/changelog-3.10.xml (hunk 2 above)
Comment (Abyss777) [inline schema/changelog-3.10.xml]: Why not

Comment (tananaev) [inline schema/changelog-3.10.xml]: There was the same issue, right? Can you please add it to this pull request.

File: schema/changelog-3.10.xml (hunk 3 above)
Comment (tananaev) [inline schema/changelog-3.10.xml]: I know, it's very confusing, so it's always better to check.

File: schema/changelog-3.10.xml [hunk 4]
```diff
@@ -62,17 +62,51 @@
...
+    </preConditions>
+    <sql>
+      CREATE TRIGGER tg_users_delete
+      ON users FOR DELETE
+      AS BEGIN
+        DELETE FROM user_user WHERE manageduserid IN (SELECT deleted.id FROM deleted)
+      END
+    </sql>
+    <sql>
+      CREATE TRIGGER tg_groups_delete
```
Comment (tananaev) [inline schema/changelog-3.10.xml]: I'm thinking if it's better to split it into `changelog-3.7-mssql`. What do you think?

Comment (Abyss777) [inline schema/changelog-3.10.xml]: Sure, it will look more logically.

**Comment by Abyss777** [comment]
All done.

It is not full replace for foreign key (there is no `onUpdate="RESTRICT"`) but good enough I think.

**Comment by Abyss777** [comment]
Or better move `changelog-3.7-mssql` to `changelog-3.7` file?

**Comment by tananaev** [comment]
Merged, I think it's fine to keep it in the same file.

**Pull request merged**